Mathlib/Probability/Distributions/Fernique.lean

```diff
@@ -0,0 +1,623 @@
+/-
+Copyright (c) 2025 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+import Mathlib.Analysis.Normed.Lp.WithLp
+import Mathlib.Analysis.SpecificLimits.ArithmeticGeometric
+import Mathlib.MeasureTheory.Constructions.BorelSpace.ContinuousLinearMap
+import Mathlib.MeasureTheory.Function.L1Space.Integrable
+import Mathlib.Topology.MetricSpace.Polish
+
+/-!
+# Fernique's theorem for rotation-invariant measures
+
+Let `μ` be a finite measure on a second-countable normed space `E` such that the product measure
+`μ.prod μ` on `E × E` is invariant by rotation of angle `-π/4`.
+Then there exists a constant `C > 0` such that the function `x ↦ exp (C * ‖x‖ ^ 2)` is integrable
+with respect to `μ`.
+
+## Sketch of the proof
+
+The main case of the proof is for `μ` a probability measure such that there exists a positive
+`a : ℝ` such that `2⁻¹ < μ {x | ‖x‖ ≤ a} < 1`. If `μ` is a probability measure and `a` does not
+exist then we can show that there is a ball with finite radius of measure 1, and the result is true
+for `C = 1` (for example), since `x ↦ exp (‖x‖ ^ 2)` is almost surely bounded.
+We then choose such an `a`.
+
+In order to show the existence of `C` such that `x ↦ exp (C * ‖x‖ ^ 2)` is integrable, we prove as
+intermediate result that for `a, c` with `2⁻¹ < c ≤ μ {x | ‖x‖ ≤ a}`,
+the integral `∫⁻ x, exp (logRatio c * a⁻¹ ^ 2 * ‖x‖ ^ 2) ∂μ` is bounded by a finite quantity that
+does not depend on `a` (`logRatio c` is a multiple of `log (c / (1 - c))`).
```

Comment:
I'm not sure why you insist that the finite quantity does not depend on `a`: since `a` is fixed, it doesn't matter anyway (all you want is the finiteness). The current formulation gave me the impression that you wanted to let a tend to infinity at some point... Same thing Line 489.

Reply:
I removed the mention from the file docstring that describes Fernique's theorem proof, but I am leaving it in the lemma docstring since it will matter for the TODO (proposition 4.16 in https://www.hairer.org/notes/SPDEs_Course.pdf)